libcovebods/tasks/checks.py

```diff
@@ -695,3 +757,1164 @@ def check_entity_statement_first_pass(self, statement):
+                            }
+                        )
+        if (
+            "recordId" in statement
+            and "recordDetails" in statement
+            and isinstance(statement["recordDetails"], dict)
+        ):
+            if "interestedParty" in statement["recordDetails"] and isinstance(
+                statement["recordDetails"]["interestedParty"], str
+            ):
+                if statement["recordDetails"]["interestedParty"] in self._records:
+                    if (
+                        self._records[statement["recordDetails"]["interestedParty"]]
+                        > self._records[statement["recordId"]]
+                    ):
+                        self._additional_check_results.append(
+                            {
+                                "type": "relationship_interested_party_not_before_relationship_in_dataset",
+                                "statement_type": None,
+                                "statement": statement.get("statementId"),
+                                "interested_party_id": statement["recordDetails"][
+                                    "interestedParty"
+                                ],
+                            }
+                        )
+
+
+class CheckStatementPersonIdentifiersHaveCorrectScheme(AdditionalCheck):
+    def __init__(self, lib_cove_bods_config, schema_object):
+        super().__init__(lib_cove_bods_config, schema_object)
+        self._count = 0
+        self._records = {}
+        self._components = {}
+
+    @staticmethod
+    def does_apply_to_schema(lib_cove_bods_config, schema_object) -> bool:
+        return schema_object.is_schema_version_equal_to_or_greater_than("0.4")
+
+    def check_person_statement_first_pass(self, statement):
+        if (
+            "recordDetails" in statement
+            and isinstance(statement["recordDetails"], dict)
+            and "identifiers" in statement["recordDetails"]
+            and isinstance(statement["recordDetails"]["identifiers"], list)
+        ):
+            for identifier in statement["recordDetails"]["identifiers"]:
+                if not (
+                    "scheme" in identifier
+                    and identifier["scheme"].count("-") == 1
+                    and len(identifier["scheme"].split("-")[0]) > 0
+                    and len(identifier["scheme"].split("-")[1]) > 0
+                ):
+                    self._additional_check_results.append(
+                        {
+                            "type": "person_identifiers_invalid_composition",
+                            "statement_type": None,
+                            "statement": statement.get("statementId"),
+                            "scheme": identifier["scheme"]
+                            if "scheme" in identifier
+                            else None,
+                        }
+                    )
+                else:
+                    other_codes = (
```

Comment:
Should this be in config, so it can be changed without needing a new release of the library? Also can there be comments explaining why we need to allow these specific other codes?

Reply:
Have moved codes into config and added comment